### ISSUE TITLE
objectives and metrics for multi-class multi-output (aka multi-label) xgboost classifier

### ISSUE THREAD

**Issue by adkinsty** [dmlc/xgboost#8289]
This [tutorial](https://xgboost.readthedocs.io/en/stable/tutorials/multioutput.html) states that XGBoost 1.6 has experimental support for multi-output classification. However, it also states that there is limited support from objectives and metrics. In the multi-class _single-output_ scenario, I use `multi:softprob` as the objective and `mlogloss` as a metric. Could I use the same objective and metric for the multi-class _multi-output_ scenario? Or are there alternative objectives and metrics that I should use instead?

**Comment by adkinsty** [comment]
Thanks for your reply. But ah, that's too bad. Do you know of a viable workaround for this? For example, could I wrap the XGBoostClassifier in sklearn's OneVsRestClassifier(), like so:

```
from sklearn.multiclass import OneVsRestClassifier
from xgboost import XGBClassifier
from sklearn.preprocessing import MultiLabelBinarizer

clf = OneVsRestClassifier(XGBClassifier(n_jobs=-1, max_depth=4))
mlb = MultiLabelBinarizer()
y = mlb.fit_transform(y)

clf.fit(X, y)
```

Forgive me if this is outside of your scope of support.

**Comment by trivialfis** [comment]
Feel free to try it. :-). We haven't have any test with that meta estimator, you are more than welcomed to add one.
